server/command_list.go

```diff
@@ -32,54 +33,78 @@ func (p *Plugin) getUpdatedInstallsForUser(userID string) ([]*Installation, erro
 		return nil, err
 	}
 
-	// cloud.Installation
+	// Grab the cloud installations belonging to this user. Note that we are not
+	// asking for deleted installations. This is done for performance reasons as
+	// we can ask for deleted installations later if necesssary.
 	cloudInstalls, err := p.cloudClient.GetInstallations(&cloud.GetInstallationsRequest{
-		OwnerID:        userID,
-		IncludeDeleted: true,
+		OwnerID:            userID,
+		IncludeGroupConfig: true,
+		IncludeDeleted:     false,
+		PerPage:            cloud.AllPerPage,
 	})
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "unable to get installations from cloud server")
 	}
 
-	// TODO: This may become a performance issue as deleted installations in the database start
-	// to grow. We could make it better by assuming that installation is not deleted. If an error
-	// is returned when asked for it, the plugin makes a follow up call to confirm that it was
-	// deleted. If deleted, the installation should be also removed from the plugin storage.
-	for _, cloudInstall := range cloudInstalls {
-		for j := 0; j < len(pluginInstalls); j++ {
-			if cloudInstall.ID == pluginInstalls[j].ID {
-				continue
-			}
-
-			if cloudInstall.DeleteAt > 0 || cloudInstall.State == cloud.ClusterInstallationStateCreationFailed {
-				err = p.deleteInstallation(pluginInstalls[j].ID)
-				if err != nil {
-					p.API.LogError(err.Error(), pluginInstalls[j].ID)
-					continue
-				}
-
-				// Notify the user that installation was removed.
-				p.PostBotDM(userID, fmt.Sprintf("Cloud installation ID %s has been removed from your Mattermost app.", pluginInstalls[j].ID))
-
-				// Update plugin installs array.
-				pluginInstalls = updatePluginInstalls(j, pluginInstalls)
-				break
+	var deleted bool
```

Comment:
Yeah, it's not necessary other than some of of our CI/CD lint+vet checks complaining when variables are shadow'ed. I couldn't remember if this was one of those repos or not so I just declared the variable outside the for loop. You are right that I could have made it compile and run the same the way you proposed otherwise.

Reply:
Not a big deal then, no worries.